edk2: fix hash

superherointj opened the pull request:
edk2: fix hash

Fixes regression:
- https://github.com/NixOS/nixpkgs/pull/321143#issuecomment-2212458679
- https://github.com/NixOS/nixpkgs/pull/325059#issuecomment-2212454476

CC @adamcstephens @RaitoBezarius

superherointj committed:
edk2: fix hash

Fixes regression:
- https://github.com/NixOS/nixpkgs/pull/321143#issuecomment-2212458679
- https://github.com/NixOS/nixpkgs/pull/325059#issuecomment-2212454476

superherointj commented:
Result of `nixpkgs-review pr 325267` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>quickemu</li>
    <li>quickgui</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF-cloud-hypervisor</li>
    <li>OVMF-cloud-hypervisor.fd</li>
    <li>OVMF.fd</li>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>edk2</li>
    <li>edk2-uefi-shell</li>
    <li>lxd-lts</li>
    <li>multipass</li>
  </ul>
</details>

superherointj commented:
Result of `nixpkgs-review pr 325267` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>quickemu</li>
    <li>quickgui</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF-cloud-hypervisor</li>
    <li>OVMF-cloud-hypervisor.fd</li>
    <li>OVMF.fd</li>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>edk2</li>
    <li>edk2-uefi-shell</li>
    <li>lxd-lts</li>
    <li>multipass</li>
  </ul>
</details>

Mic92 commented on pkgs/by-name/ed/edk2/package.nix:
```diff
@@ -51,7 +51,9 @@ edk2 = stdenv.mkDerivation {
-    hash = "sha256-7vNodHocwqQiO0ZXtqo8lEOFyt8JkFHcAathEhrKWE0=";
+    # Note: Hash changes whenever OpenSSL package is upgraded,
+    #       whenever upgrading edk2 check for concurrent OpenSSL upgrades
+    hash = "sha256-4JDeA4QBa/FYqgHi5l6FT9EdcTtFvIbKzO+ywT1BsC8=";
```
uff. That's bad. We should create a second derivation that takes this fixed-input derivation as an input and puts a new patched source tree.

Mic92 replied:
Going to push an improved version.

Mic92 replied:
Fixed.

Mic92 committed:
edk2: reformat with nixfmt

Mic92 committed:
edk2: use postPatch to de-vendor openssl source

using postFetch in a fixed-ouput derivation is unsound because it will
break on each openssl update

Mic92 commented:
Sorry for the reformat commit but the original formatting was not very readable.

RaitoBezarius commented:
So is OVMF using the debundled OpenSSL?

adamcstephens commented:
> So is OVMF using the debundled OpenSSL?

Yes, this appears to be a revert of the non-upgrade changes from my PR, so removal of bundled openssl is a postPatch again

vcunat commented:
Has anyone tried `nixosTests.boot.uefiUsb`?  (I might have some issues here)
@ofBorg build nixosTests.boot.uefiUsb

mjoerg reviewed (commented):
I think this still uses the bundled OpenSSL version in `edk2.mkDerivation`.  I gave it a try, and added a custom `updateScript`: https://github.com/mjoerg/nixpkgs/commit/edk2-fix-hash/

superherointj commented:
> Has anyone tried `nixosTests.boot.uefiUsb`? (I might have some issues here) @ofborg build nixosTests.boot.uefiUsb

At `master`:

![image](https://github.com/NixOS/nixpkgs/assets/5861043/68c3135d-57e0-4bbb-a9fc-3e1384f2bf11)

At `superherointj:edk2-fix-hash`:

![image](https://github.com/NixOS/nixpkgs/assets/5861043/53cdb2a2-efbb-456e-bca5-7caa32620488)

`timeout reached; test terminating...`

adamcstephens reviewed (changes requested):
Changing review as this does not properly devendor OpenSSL for edk2.mkDerivation which is a requirement.

adamcstephens commented:
There was a bit too much going on here, so I opened https://github.com/NixOS/nixpkgs/pull/325403 which is more of a direct revert back to the previous state.
